core/src/com/unciv/ui/pickerscreens/ImprovementPickerScreen.kt

```diff
@@ -58,8 +61,23 @@ class ImprovementPickerScreen(val tileInfo: TileInfo, val onAccept: ()->Unit) :
 
             improvementButtonTable.add(image).size(30f).pad(10f)
 
+            // allow multiple key mappings to technologically supersede each other
+            var shortcutKey = improvement.shortcutKey
+            if (shortcutKey != null) {
+                val techLevel = ruleSet.technologies[improvement.techRequired]?.column?.columnNumber ?: -1
```

Comment:
Since you're using this both here and below, I would add a short `fun getTechColumn(improvement:Improvement)` above so A. it's immediately obvious we're using the same info to compare the two and B. to ensure that this doesn't go out-of-synch in the future somehow

Reply:
True again - I probably considered a simple additional property/function on Technology or RuleSet and ditched it to keep this within a single code file - wait a minute... Looks like this didn't merge properly, it shouldn't touch TileInfo, those changes belong to #3866 - I based this branch on that branch and local commits and pushes looked OK. Sigh. So github doesn't preserve based-on info for a branch... Still, the hash of the duplicated commit is identical so the result should be OK, only the "Files changed" tab misleads a little?